metricbeat/module/elasticsearch/shard/data_xpack.go

```diff
@@ -0,0 +1,72 @@
+package shard
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/module/elasticsearch"
+)
+
+func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, content []byte) {
+	stateData := &stateStruct{}
+	err := json.Unmarshal(content, stateData)
+	if err != nil {
+		r.Error(err)
+		return
+	}
+
+	logp.Err("Content: %s", content)
+	logp.Err("URL: %s", m.HostData().SanitizedURI+statePath)
+
+	nodeInfo, err := elasticsearch.GetNodeInfo(m.HTTP, m.HostData().SanitizedURI+statePath, stateData.MasterNode)
+	if err != nil {
+		r.Error(err)
+		return
+	}
+
+	// This is currently needed because the cluser_uuid is `na` in stateData in case not the full state is requested.
+	// This needs to be brought up with the ES team.
+	clusterID, err := elasticsearch.GetClusterID(m.HTTP, m.HostData().SanitizedURI+statePath, stateData.MasterNode)
```

Comment:
@pickypg @tlrx You might now more here? On ES master the following request `http://localhost:9200/_cluster/state/version,master_node,routing_table?pretty` returns:

```
{
  "cluster_name" : "elasticsearch",
  "compressed_size_in_bytes" : 36170,
  "cluster_uuid" : "_na_",
  "version" : 59,
```

If I fetch `http://localhost:9200/_cluster/state/?pretty` I get

```
{
  "cluster_name" : "elasticsearch",
  "compressed_size_in_bytes" : 36170,
  "cluster_uuid" : "Ir3Dhzo-TluhYWEaz4UZlA",
  "version" : 59,
```

Is this a bug?